From b39f5f2f4a3c029d281f7e98ef3a3f39e5a328c2 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Lothar=20Wa=C3=9Fmann?= <LW@KARO-electronics.de>
Date: Tue, 21 Jan 2020 12:56:46 +0100
Subject: [PATCH 31/35] imx8mm: clock: fix setting for PHY_REF_CLK at 25MHz

The ENET_PHY__REF_CLK_ROOT mux on the i.MX8MMini does not provide a
25MHz clock setting. Rename the corresponding definition to reflect
the actually selectable 50MHz clock and set the POST_DIV to 2 to
achieve the 25MHz ref clock.
---
 arch/arm/include/asm/arch-imx8m/clock_imx8mm.h | 2 +-
 arch/arm/mach-imx/imx8m/clock_imx8mm.c         | 4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/arch/arm/include/asm/arch-imx8m/clock_imx8mm.h b/arch/arm/include/asm/arch-imx8m/clock_imx8mm.h
index ac7590a..a5db919 100644
--- a/arch/arm/include/asm/arch-imx8m/clock_imx8mm.h
+++ b/arch/arm/include/asm/arch-imx8m/clock_imx8mm.h
@@ -943,7 +943,7 @@ enum dram_bypassclk_val {
 #define ENET_AXI_CLK_ROOT_FROM_PLL_SYS_PFD4_CLK			0x07000000
 #define ENET_AXI_CLK_ROOT_FROM_SYS1_PLL_266M			0x01000000
 #define ENET1_TIME_CLK_ROOT_FROM_PLL_ENET_MAIN_100M_CLK		0x01000000
-#define ENET_PHY_REF_CLK_ROOT_FROM_PLL_ENET_MAIN_25M_CLK	0x01000000
+#define ENET_PHY_REF_CLK_ROOT_FROM_PLL_ENET_MAIN_50M_CLK	0x01000000
 
 enum enet_freq {
 	ENET_25MHZ = 0,
diff --git a/arch/arm/mach-imx/imx8m/clock_imx8mm.c b/arch/arm/mach-imx/imx8m/clock_imx8mm.c
index 92fc8a7..5f58424 100644
--- a/arch/arm/mach-imx/imx8m/clock_imx8mm.c
+++ b/arch/arm/mach-imx/imx8m/clock_imx8mm.c
@@ -795,9 +795,9 @@ int set_clk_enet(enum enet_freq type)
 
 #ifdef CONFIG_FEC_MXC_25M_REF_CLK
 	target = CLK_ROOT_ON |
-		 ENET_PHY_REF_CLK_ROOT_FROM_PLL_ENET_MAIN_25M_CLK |
+		 ENET_PHY_REF_CLK_ROOT_FROM_PLL_ENET_MAIN_50M_CLK |
 		 CLK_ROOT_PRE_DIV(CLK_ROOT_PRE_DIV1) |
-		 CLK_ROOT_POST_DIV(CLK_ROOT_POST_DIV1);
+		 CLK_ROOT_POST_DIV(CLK_ROOT_POST_DIV2);
 	clock_set_target_val(ENET_PHY_REF_CLK_ROOT, target);
 #endif
 	/* enable clock */
-- 
2.7.4

